db-async-common/src/main/java/com/github/jasync/sql/db/pool/ActorBasedObjectPool.kt

```diff
@@ -496,6 +498,10 @@ private class ObjectPoolActor<T : PooledObject>(
         when (tried) {
             is Failure -> throw tried.exception
         }
+        val age = System.currentTimeMillis() - a.creationTime
+        if (configuration.maxTtl > 0 && age > configuration.maxTtl) {
+            throw ObjectAgedOutException(a, age, configuration.maxTtl)
```

Comment:
I don't think we should throw an exception in case of ttl passed, just destroy the item on validation

Reply:
Should validate() call dispose directly then?

Reply:
Calling `item.destroy()` directly, let me know if that's not what you had in mind.

Reply:
Looking at the code again it seems that you did the right thing. The only comment then is to call the exception MaxTtlPassedException

Reply:
:+1: